src/main/scala/inox/ast/SymbolOps.scala

```diff
@@ -115,42 +116,30 @@ trait SymbolOps { self: TypeOps =>
     case _ => false
   }
 
-  /** Returns true when the given expression is ground, pure, and does not contain
-   *  choose or quantifiers, and can thus be simplified.
-   */
-  def canBeForced(expr: Expr): Boolean = {
-    isGround(expr) && isPure(expr)(PurityOptions.AssumeChecked) && !containsChooseOrQuantifiers(expr)
-  }
-
   /** Simplify all the pure and ground sub-expressions of the given expression,
-   * by forcing them using [[inox.evaluators.Evaluator]].
-   *
-   * @see [[canBeForced]]
-   */
-  def simplifyGround(expr: Expr)(implicit sem: symbols.Semantics, ctx: Context): Expr = {
-    val evaluator = sem.getEvaluator
+    * by evaluating them using [[inox.evaluators.Evaluator]].
+    * If `force` is omitted, the given expression will only be evaluated if it is
+    * ground, pure, and does not contain choose or quantifiers.
+    */
+  def simplifyGround(expr: Expr, force: Boolean = false)
+                    (implicit sem: symbols.Semantics, ctx: Context, opts: PurityOptions): Result[Expr] = {
```

Comment:
Ah I didn't realize you needed the context for getting the evaluator. You can get the PurityOptions from the context as `PurityOptions(ctx)`.